packages/twenty-utils/congratulate-dangerfile.ts

```diff
@@ -0,0 +1,72 @@
+import { danger } from 'danger';
+
+const ordinalSuffix = (number) => {
+  const v = number % 100;
+  if (v === 11 || v === 12 || v === 13) {
+    return number + 'th';
+  }
+  const suffixes = { 1: 'st', 2: 'nd', 3: 'rd' };
+  return number + (suffixes[v % 10] || 'th');
+};
+
+const fetchContributorStats = async (username: string) => {
+  const apiUrl = `https://twenty.com/api/contributors/contributorStats/${username}`;
+
+  const response = await fetch(apiUrl);
+  const data = await response.json();
+  return data;
+};
+
+const fetchContributorImage = async (username: string) => {
+  const apiUrl = `https://twenty.com/api/contributors/${username}/og.png`;
+
+  await fetch(apiUrl);
+};
+
+const runCongratulate = async () => {
+  const pullRequest = danger.github.pr;
+  const userName = pullRequest.user.login;
+
+  const { data: pullRequests } =
+    await danger.github.api.rest.search.issuesAndPullRequests({
+      q: `is:pr author:${userName} is:closed repo:twentyhq/twenty`,
+      per_page: 2,
+      page: 1,
+    });
+
+  let stats;
+  const isFirstPR = pullRequests.total_count === 1;
+
+  if (!isFirstPR) {
+    stats = await fetchContributorStats(userName);
+  } else {
+    stats = { mergedPRsCount: 1, rank: 52 };
+  }
+
+  const contributorUrl = `https://twenty.com/contributors/${userName}`;
+
+  // Pre-fetch to trigger cloudflare cache
+  await fetchContributorImage(userName);
+
+  const message =
+    `Thanks @${userName} for your contribution!\n` +
+    `This marks your **${ordinalSuffix(
+      stats.mergedPRsCount,
+    )}** PR on the repo. ` +
+    `You're **top ${stats.rank}%** of all our contributors 🎉\n` +
```

Comment:
Maybe hide this is if it's their first contribution at least? Feels a bit strange otherwise

Reply:
Agreed !